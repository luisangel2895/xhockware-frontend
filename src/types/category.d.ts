export interface Category {
  name: string;
  status: boolean;
}

export interface AllCategories {
  all: {
    name: string;
    status: boolean;
  };
  health: {
    name: string;
    status: boolean;
  };
  economy: {
    name: string;
    status: boolean;
  };
  tecnology: {
    name: string;
    status: boolean;
  };
  politics: {
    name: string;
    status: boolean;
  };
  ecology: {
    name: string;
    status: boolean;
  };
  social: {
    name: string;
    status: boolean;
  };
}
