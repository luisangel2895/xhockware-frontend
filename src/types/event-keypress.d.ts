export interface EventKeyPress {
  isTrusted: boolean;
  altKey: boolean;
  bubbles: boolean;
  cancelBubble: boolean;
  cancelable: boolean;
  charCode: number;
  code: string;
}
