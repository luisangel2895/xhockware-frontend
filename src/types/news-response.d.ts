export interface NewsResponse {
  status: string;
  totalResults: number;
  articles: News[];
}

export interface News {
  source: Source;
  author: string;
  category: string;
  title: string;
  description: string;
  url: string;
  urlToImage: string;
  publishedAt: string;
  content: string;
}

export interface Source {
  id: string;
  name: string;
}
