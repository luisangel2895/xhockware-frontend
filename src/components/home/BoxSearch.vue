<template lang="pug">
.box-search
	img(src="@/assets/extra-icons/search.png").box-search__icon-search
	input(type="text" placeholder="Search").box-search__content
	img(src="@/assets/extra-icons/mic.png").box-search__icon-mic

</template>
<script lang="ts">
import { Options, Vue } from "vue-class-component";

@Options({})
export default class BoxSearch extends Vue {}
</script>

<style lang="scss" scoped>
.box-search {
  width: 100%;
  max-width: 600px;
  display: flex;
  padding: 15px;
  border-radius: 10px;
  background-color: $grey_lighter;
  margin-top: 30px;

  &__icon-search {
    cursor: pointer;
  }

  &__content {
    width: 100%;
    margin: 0px 10px 0px 10px;
    font-family: $medium;
    font-size: 1.6rem;
    color: $grey_primary;
    border: none;
    background-color: $grey_lighter;
    &:focus {
      outline: none;
    }
    &::placeholder {
      color: $grey_primary;
    }
  }
}
</style>
