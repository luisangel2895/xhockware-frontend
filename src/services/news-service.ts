import axios from "axios";

import { News } from "@/types/news-response";

export const urlBaseNews = "http://localhost:8000/v1/news";
export const defaultWord = "";

export const getNewsService = async (
  word: string = defaultWord
): Promise<News[] | never> => {
  try {
    const { data } = await axios.get(`${urlBaseNews}`, {
      params: {
        q: word,
      },
    });
    return data.articles;
  } catch (e) {
    throw new Error("sorry we couldn't bring the news");
  }
};
