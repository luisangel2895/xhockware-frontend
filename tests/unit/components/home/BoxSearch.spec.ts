import { shallowMount } from "@vue/test-utils";
import { createStore } from "vuex";
// Components
import BoxSearch from "@/components/home/BoxSearch.vue";
// Services
import { getNewsService } from "@/services/news-service";
// Types
import { News } from "@/types/news-response";
import { AllCategories } from "@/types/category";

interface State {
  news: News[];
  categories: AllCategories;
}

const store = createStore({
  state(): State {
    return {
      news: [],
      categories: {
        all: {
          name: "all",
          status: true,
        },
        health: {
          name: "health",
          status: false,
        },
        economy: {
          name: "economy",
          status: false,
        },
        tecnology: {
          name: "tecnology",
          status: false,
        },
        politics: {
          name: "politics",
          status: false,
        },
        ecology: {
          name: "ecology",
          status: false,
        },
        social: {
          name: "social",
          status: false,
        },
      } as AllCategories,
    };
  },
  getters: {
    getCategories: (state) => (state as State).categories,
  },
  mutations: {
    SET_TOTAL_NEWS(state, payload: News[]) {
      (state as State).news = payload;
    },
  },
  actions: {
    async getNews({ commit }, payload: string) {
      const news: News[] = await getNewsService(payload);
      if (!payload) {
        commit("SET_TOTAL_NEWS", news);
      }
    },
  },
});

const wrapper = shallowMount(BoxSearch, {
  global: {
    plugins: [store],
    stubs: ["font-awesome-icon"],
    mocks: {
      $t: () => "search",
    },
  },
});

describe("Test in component BoxSearch.vue", (): void => {
  it("Match with snapshot", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
  it("Components (font-awesome-icon, input, img", (): void => {
    expect(wrapper.find("font-awesome-icon-stub").exists()).toBeTruthy();
    expect(wrapper.find("input").exists()).toBeTruthy();
    expect(wrapper.find("img").exists()).toBeTruthy();
  });
  it("Loading by default the placeholder in english", async (): Promise<void> => {
    expect(wrapper.find("input").attributes("placeholder")).toBe("search");
  });
  it("Called action (getNews) when clicked search icon searching", async (): Promise<void> => {
    const cleanCategoriesSpy = jest.spyOn<any, any>(
      BoxSearch.methods,
      "cleanCategories"
    );
    const wrapper = shallowMount(BoxSearch, {
      global: {
        plugins: [store],
        stubs: ["font-awesome-icon"],
        mocks: {
          $t: () => "search",
        },
      },
    });
    const icon_search = wrapper.find("font-awesome-icon-stub");
    await icon_search.trigger("click");
    expect(cleanCategoriesSpy).toHaveBeenCalled();
  });
  it("Called action (getNews) when press Enter key", async (): Promise<void> => {
    const cleanCategoriesSpy = jest.spyOn<any, any>(
      BoxSearch.methods,
      "cleanCategories"
    );
    const wrapper = shallowMount(BoxSearch, {
      global: {
        plugins: [store],
        stubs: ["font-awesome-icon"],
        mocks: {
          $t: () => "search",
        },
      },
    });
    const icon_search = wrapper.find("input");
    await icon_search.trigger("keypress", { code: "Enter" });
    expect(cleanCategoriesSpy).toHaveBeenCalled();
  });
});
