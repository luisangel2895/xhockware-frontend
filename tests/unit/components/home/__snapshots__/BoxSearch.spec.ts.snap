// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in component BoxSearch.vue Match with snapshot 1`] = `
<div class="box-search">
  <font-awesome-icon-stub class="box-search__icon-search" icon="fa,magnifying-glass"></font-awesome-icon-stub><input class="box-search__content" type="text" placeholder="search"><img class="box-search__icon-mic">
</div>
`;
